src/algebra/module/pi.lean

```diff
@@ -33,6 +33,16 @@ lemma smul_apply' {g : I → Type*} [∀ i, has_scalar (f i) (g i)] (s : Π i, f
   (s • x) i = s i • x i :=
 rfl
 
+instance is_scalar_tower {g : I → Type*} {α : Type*}
+  [Π i, has_scalar α $ f i] [Π i, has_scalar (f i) (g i)] [Π i, has_scalar α $ g i]
+  [Π i, is_scalar_tower α (f i) (g i)] : is_scalar_tower α (Π i : I, f i) (Π i : I, g i) :=
+⟨λ x y z, funext $ λ i, smul_assoc x (y i) (z i)⟩
+
+instance is_scalar_tower' {g : I → Type*} {h : I → Type*}
+  [Π i, has_scalar (f i) (g i)] [Π i, has_scalar (g i) (h i)] [Π i, has_scalar (f i) (h i)]
+  [Π i, is_scalar_tower (f i) (g i) (h i)] : is_scalar_tower (Π i, f i) (Π i, g i) (Π i, h i) :=
+⟨λ x y z, funext $ λ i, smul_assoc (x i) (y i) (z i)⟩
+
```

Comment:
Why not include `is_scalar_tower α β (Π i : I, f i)`?

Reply:
Done.